baseband/vlbi_base/base.py

```diff
@@ -188,19 +203,19 @@ def tell(self, unit=None):
         if unit == 'time':
             return self.start_time + self.tell(unit=u.s)
 
-        return (self.offset * u_sample).to(unit, equivalencies=[(u.s, u.Unit(
-            self.samples_per_frame * self.frames_per_second * u_sample))])
+        return (self.offset / self.sample_rate).to(unit)
 
     def _frame_info(self):
         offset = (self.offset +
                   self.header0['frame_nr'] * self.samples_per_frame)
+        framerate = self.sample_rate.to_value(u.Hz) / self.samples_per_frame
         full_frame_nr, extra = divmod(offset, self.samples_per_frame)
-        dt, frame_nr = divmod(full_frame_nr, self.frames_per_second)
-        return int(dt), int(frame_nr), extra
+        dt, frame_nr = divmod(full_frame_nr, framerate)
+        return int(dt), int(frame_nr), int(extra)
 
     def __repr__(self):
         return ("<{s.__class__.__name__} name={s.name} offset={s.offset}\n"
-                "    frames_per_second={s.frames_per_second},"
+                "    sample_rate={s.sample_rate:.6g},"
```

Comment:
I would not do the `.6g` here since I think generally the numbers will be OK. Only really necessary for GSB.